src/node_crypto.cc

```diff
@@ -5925,6 +5943,27 @@ void GenerateKeyPairEC(const FunctionCallbackInfo<Value>& args) {
   GenerateKeyPair(args, 2, std::move(config));
 }
 
+void GenerateKeyPairEdDSA(const FunctionCallbackInfo<Value>& args) {
+#ifdef EVP_PKEY_ED25519
+  CHECK(args[0]->IsString());
+  String::Utf8Value curve_name(args.GetIsolate(), args[0].As<String>());
+  int id;
+  if (strcmp(*curve_name, "ed25519") == 0) {
+    id = EVP_PKEY_ED25519;
+  } else {
+    CHECK_EQ(strcmp(*curve_name, "ed448"), 0);
+    id = EVP_PKEY_ED448;
+  }
```

Comment:
For efficiency reasons you could pass in `type === 'ed25519'` in JS land and do:
```c++
const int id = args[0]->IsTrue() ? EVP_PKEY_ED25519 : EVP_PKEY_ED448;
```